### PR TITLE
Handle build and run errors

### PR DESCRIPTION
**Pull request by tcstewar**
Exceptions thrown during the build process and when running the model should now be caught by nengo_gui and displayed to the user.

### CODE REVIEW

File: nengo_gui/page.py [hunk 1]
```diff
@@ -416,11 +416,21 @@ def build(self):
                 old_sim.sim = None
                 old_sim.finished = True
 
+            patch = nengo_gui.exec_env.ExecutionEnvironment(self.filename,
```
Comment (Seanny123) [inline nengo_gui/page.py]: Why is this variable called `patch`?

Comment (tcstewar) [inline nengo_gui/page.py]: Oops, good catch -- that's a holdover from the old name.  Changed to ```exec_env``` now.  :)

**Comment by Seanny123** [comment]
LGTM, from a syntax and style basis. I don't know the code well enough to know if there was a more elegant way of accomplishing this, so it's up to you if you want to wait for someone more familiar with the code base.

**Comment by tcstewar** [comment]
I just caught a debug message I left in there...  so I'll give it another day or so before merging.  :)  But I'm pretty happy with this code organization.

**Pull request merged**